### PR TITLE
[12.0][IMP] mail: fast fill add_sign in mail.message

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
If you have a lot of messages, filling this new column `add_sign` take much time.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by AaronHForgeFlow** [comment]
always True?

**Comment by MiquelRForgeFlow** [comment]
> always True?

By default, it's always True.

**Comment by MiquelRForgeFlow** [comment]
I don't know if this PR provides a faster way. For example, in a database I used, the query lasted **19 minutes** to finally show the `OpenUpgrade: 8397417 rows affected` message.

**Comment by pedrobaeza** [comment]
What about trying to put on pre- the `ALTER TABLE ... ADD...` with default value to see if it's faster?

**Comment by MiquelRForgeFlow** [comment]
Thanks, it solved the problem that way. It was nearly an instant :)

File: addons/mail/migrations/12.0.1.0/pre-migration.py [hunk 1]
```diff
@@ -14,8 +14,20 @@
 ]
 
 
+def fill_mail_message_add_sign_default(env):
+    """Fast way to fill this new field"""
+    if not openupgrade.column_exists(env.cr, 'mail_message', 'add_sign'):
```
Comment (pedrobaeza) [inline addons/mail/migrations/12.0.1.0/pre-migration.py]: Why this condition?

File: addons/mail/migrations/12.0.1.0/pre-migration.py [hunk 2]
```diff
@@ -14,8 +14,20 @@
...
+        openupgrade.logged_query(
+            env.cr, """
+            ALTER TABLE mail_message
+            ADD COLUMN add_sign boolean
+            DEFAULT TRUE""",
```
Comment (pedrobaeza) [inline addons/mail/migrations/12.0.1.0/pre-migration.py]: Doubt I have now is this default will act for new additions. I think you must remove that default after this one.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: addons/mail/migrations/12.0.1.0/openupgrade_analysis_work.txt [hunk 3]
```diff
@@ -46,6 +46,8 @@ mail         / mail.channel             / moderator_ids (many2many)     : NEW re
 # NOTHING TO DO
 
 mail         / mail.message             / add_sign (boolean)            : NEW
+# DONE: pre-migration: filled this new field using sql query for fast migration
```
Comment (pedrobaeza) [inline addons/mail/migrations/12.0.1.0/openupgrade_analysis_work.txt]: s/fast/faster

File: addons/mail/migrations/12.0.1.0/pre-migration.py (hunk 2 above)
Comment (MiquelRForgeFlow) [reply addons/mail/migrations/12.0.1.0/pre-migration.py]: how?

Comment (pedrobaeza) [reply addons/mail/migrations/12.0.1.0/pre-migration.py]: https://popsql.com/learn-sql/postgresql/how-to-remove-a-default-value-from-a-column-in-postgresql/

Comment (MiquelRForgeFlow) [reply addons/mail/migrations/12.0.1.0/pre-migration.py]: thanks, done

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**